kinto_megaphone/listeners.py

```diff
@@ -49,23 +68,31 @@ def filter_records(self, impacted_records):
             record = delta['new']
             record_bucket = record['bucket']
             record_collection = record['collection']
-            for (resource_name, matchdict) in self.resources:
-                if resource_name == 'bucket':
-                    resource_bucket = matchdict['id']
-                else:
-                    resource_bucket = matchdict.get('bucket_id')
-
-                if resource_name == 'collection':
-                    resource_collection = matchdict['id']
-                else:
+
+            match = False
+
+            for (resource_name, matchdict) in self.included_resources:
+                resource_bucket = matchdict['bucket_id']
+                resource_collection = matchdict.get('collection_id')
+
+                if resource_name == 'bucket' and resource_bucket == record_bucket:
+                    match = True
+
+                if resource_collection and resource_collection == record_collection:
+                    match = True
+
+            if match:
+                for (resource_name, matchdict) in self.excluded_resources:
+                    resource_bucket = matchdict['bucket_id']
                     resource_collection = matchdict.get('collection_id')
 
-                if resource_bucket and resource_bucket != record_bucket:
-                    continue
+                    if resource_name == 'bucket' and resource_bucket == record_bucket:
+                        match = False
 
-                if resource_collection and resource_collection != record_collection:
-                    continue
+                    if resource_collection and resource_collection == record_collection:
+                        match = False
```

Comment:
It might be good to extract a utility function for checking if one of these transformed resources matches some object.